src/test/java/jp/co/yahoo/yosegi/message/formatter/json/TestJacksonArrayFormatter.java

```diff
@@ -0,0 +1,358 @@
+  }
+
+  @Test
+  public void T_write_3() throws IOException {
+    JacksonArrayFormatter formatter = new JacksonArrayFormatter();
+    JsonNode node = formatter.write( null );
+    assertTrue( ( node instanceof ArrayNode ) );
+
+    ArrayNode arrayNode = (ArrayNode)node;
+    assertEquals( arrayNode.size() , 0 );
+  }
+
+  @Test
+  public void T_write_4() throws IOException {
+    List<Object> list = Arrays.asList( Arrays.asList( "a" , "b" , "c" ) );
+    JacksonArrayFormatter formatter = new JacksonArrayFormatter();
+    JsonNode node = formatter.write( list );
+    assertTrue( ( node instanceof ArrayNode ) );
+
+    ArrayNode parentNode = (ArrayNode)node;
+    assertEquals( parentNode.size() , 1 );
+
+    JsonNode childNode = parentNode.get(0);
+    assertTrue( ( childNode instanceof ArrayNode ) );
+    ArrayNode arrayNode = (ArrayNode)childNode;
+
+    assertEquals( arrayNode.size() , 3 );
+    assertTrue( arrayNode.get(0).isTextual() );
+    assertTrue( arrayNode.get(1).isTextual() );
+    assertTrue( arrayNode.get(2).isTextual() );
+    assertEquals( arrayNode.get(0).asText() , "a" );
+    assertEquals( arrayNode.get(1).asText() , "b" );
+    assertEquals( arrayNode.get(2).asText() , "c" );
+  }
+
+  @Test
+  public void T_write_5() throws IOException {
+    Map<Object,Object> child = new HashMap<Object,Object>();
+    child.put( "key1" , "a" );
+    child.put( "key2" , "b" );
+    child.put( "key3" , "c" );
+    List<Object> list = Arrays.asList( child );
+    JacksonArrayFormatter formatter = new JacksonArrayFormatter();
+    JsonNode node = formatter.write( list );
+    assertTrue( ( node instanceof ArrayNode ) );
+
+    ArrayNode parentNode = (ArrayNode)node;
+    assertEquals( parentNode.size() , 1 );
+
+    JsonNode childNode = parentNode.get(0);
+    assertTrue( ( childNode instanceof ObjectNode ) );
+    ObjectNode objNode = (ObjectNode)childNode;
+
+    assertEquals( objNode.size() , 3 );
+    assertTrue( objNode.get("key1").isTextual() );
+    assertTrue( objNode.get("key2").isTextual() );
+    assertTrue( objNode.get("key3").isTextual() );
+    assertEquals( objNode.get("key1").asText() , "a" );
+    assertEquals( objNode.get("key2").asText() , "b" );
+    assertEquals( objNode.get("key3").asText() , "c" );
+  }
+
+  @Test
+  public void T_writeParser_1() throws IOException {
+    JacksonArrayFormatter formatter = new JacksonArrayFormatter();
+    JsonNode node = formatter.writeParser( new TestPrimitiveParser( true ) );
+    assertTrue( ( node instanceof ArrayNode ) );
+
+    ArrayNode arrayNode = (ArrayNode)node;
+    assertEquals( arrayNode.size() , 3 );
+    assertTrue( arrayNode.get(0).isTextual() );
+    assertTrue( arrayNode.get(1).isTextual() );
+    assertTrue( arrayNode.get(2).isTextual() );
+    assertEquals( arrayNode.get(0).asText() , "a" );
+    assertEquals( arrayNode.get(1).asText() , "b" );
+    assertEquals( arrayNode.get(2).asText() , "c" );
+  }
+
+  @Test
+  public void T_writeParser_2() throws IOException {
```

Comment:
Please write the test method name is to describe what is wanted to in the test instead of using number (it makes vague one).

Reply:
I changed the method name.